COMPONENT_GUIDE.md

```diff
@@ -1,317 +1,4 @@
-- [Globally available components](#globally-available-components)
-  - [Video](#video)
-    - [Usage](#usage)
-  - [Intro](#intro)
-    - [Usage](#usage-1)
-  - [Steps](#steps)
-    - [Usage](#usage-2)
-  - [Step](#step)
-    - [Usage](#usage-3)
-- [A code example](#a-code-example)
-- [Another code example](#another-code-example)
-  - [Code blocks](#code-blocks)
-    - [Usage](#usage-4)
-  - [Callouts](#callouts)
-    - [Usage](#usage-5)
-  - [Related Resources](#related-resources)
-    - [Maximum references](#maximum-references)
-    - [Resource population logic](#resource-population-logic)
-    - [Order of priority](#order-of-priority)
-    - [Resource site tags](#resource-site-tags)
-  - [Announcement Banner](#announcement-banner)
+# Component Documentation
 
-# Globally available components
-
-## Video
-
-`<Video />` component provides formatting for videos in markdown.
-
-### Usage
-
-The Video component requires two props:
-
-- `id`: the video ID
-- `type`: the host of the video. Accepted values are:
-  - `wistia`
-  - `youtube`
-
-```md
-<Video id="zxunt1u1as" type="wistia"/>
-```
-
-## Intro
-
-The `<Intro />` component provides formatting for the title and introduction of the markdown document.
-
-### Usage
-
-It takes the title provided in the Frontmatter and accepts plain text for the description. An example of Frontmatter that will have a title of **Example Guide**:
-
-```
----
-path: '/example'
-duration: '30 min'
-title: 'Example Guide'
-template: 'GuideTemplate'
-description: 'Example guide page'
----
-```
-
-It also accepts a `<Video />` component as a child, which it will place on the left side of the description.
-
-```md
-<Intro>
-This is a description for the markdown guide.
-
-<Video id="zxunt1u1as" type="youtube"/>
-</Intro>
-```
-
-If there is a more than plain text and a `<Video />` (such as a code snippet or another component) the content will be posted on the left side below the description.
-
-## Steps
-
-The `<Steps />` is a required container for the individual `<Step />` components and will autonumber from top to bottom.
-
-### Usage
-
-The Steps component accepts `<Step/>` components as its children and will increment by the number of child components.
-
-```md
-<Steps>
```

Comment:
`<Step>` and `<Steps>` are only available on the docs site and not in the theme. We should proabably keep this around.

Reply:
same question as above. :)

Reply:
I'll answer here for both 😄 

The reason is because these are patterns that are specific to the dev site and are not used by other sites. Our philosophy so far has been: Only put things in the theme that we know will be shared amongst 2 or more sites. This allows us to flush out design and UX before we just share amongst all the sites. Its expensive to make changes to the theme, and we also want to avoid breaking changes there, since it is versioned. Its much easier to make a breaking change in the site itself than in the theme.

Specifically, here are the behaviors that we may not want to pull over for each of these 2 components:

* `Intro`

This component does a couple things:

1) Enlarges the text a bit and changes its text color to make it standout among the regular body text
2) Auto-detects images/videos in the intro and automatically puts them in a right column.

Its 2) that worries me in the theme. While this makes a lot of sense for the dev site, where its very guide oriented, this pattern may or may not work for the docs site (no docs site pages uses this kind of pattern). If we were to extract this to the theme, we'd likely want to take out the behavior to do the auto columns. And if we do that, then it means we'd need to customize that component anyways if we pulled it back into the dev site.

For now, we'd rather just iterate on this in the dev site until we have a need for this behavior across all sites for their intros.

* `Step`/`Steps`

These components are very guide oriented and much better suit the patterns on the dev site instead of the docs site. We may decide we like this pattern and want to bring it to all the sites, but in doing so, I'd recommend a rename ([its currently confusing](https://github.com/newrelic/developer-website/issues/726)).

These components are also packed with all of the functionality from the recent innovation week. That work may or may not be specific to the dev site, so if we extract, we'd want to make decisions on how we can best make sure that functionality can scale across all use cases. Again, I'd like to wait to understand the use case on say, the docs site, before we fully extract this so that we make sure its API covers the cases we need without needing to make breaking changes.

Hopefully that helps :)

Reply:
make sense! I'll add these back to this file, have we documented this _philosophy_ anywhere? this may be something useful to put into our Documentation of the theme.

Reply:
> have we documented this philosophy anywhere

😅 ...... no